src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -306,3 +310,238 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
+    self._sas_admin.TriggerLoadDpas()
+
+    # STEP 2 : ESC inf rms SAS about inactive DPA
+    self._sas_admin.TriggerBulkDpaActivation({'activate': False})
+
+    # Step 3 : creates multiple SAS TH, and Load predefined FAD,CBSD
+    for test_harness in config['sasTestHarnessConfigs']:
+      # Initialize SAS Test Harness Server instance to dump FAD records
+      sas_test_harness_object = SasTestHarnessServer(test_harness['sasTestHarnessName'],
+                                                           test_harness['hostName'],
+                                                           test_harness['port'],
+                                                           test_harness['serverCert'],
+                                                           test_harness['serverKey'],
+                                                           test_harness['caCert'])
+      # Start the server
+      sas_test_harness_object.start()
+
+      # informing SAS UUT about SAS Test Harnesses
+      certificate_hash = getCertificateFingerprint(test_harness['serverCert'])
+      self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                     'url': sas_test_harness_object.getBaseUrl()})
+      sas_test_harness_objects.append(sas_test_harness_object)
+   
+    # Step 4,5 : Inject IAP protected entities into UUT
+    for iteration_content in config['iterationData']:
+
+      # Execute steps for single iteration
+      self.executeSingleMCPIteration(test_type, iteration_content, sas_test_harness_objects, 
+                                  domain_proxy_objects)
+
+    # Stopping Test harness servers
+    for test_harness in sas_test_harness_objects:
+      test_harness.shutdown()
+      del test_harness
+
+  def checkMcpIap(self, protected_entity, agg_interference_list, iap_refmodel_margin_list, delta_iap):
+    """Calculating IAP for each FSS and ESC sensor protection point,
+    and for at least 95% of the protection points of each PPA and GWPZ.
+    """
+    pass
+
+  def callAggregateInterferenceModel(self, cbsd_list, protected_entity):
+    """Args:
+         cbsd_list: List of cbsds with Authorized grants
+         protected_entity: A dictionary with multiple key-value pairs where
+         keys are protection entities
+
+       Returns:
+         Aggregate Interference list
+    """
+    pass
+
+  def executeSingleMCPIteration(self, test_type, iteration_content, sas_test_harness_objects,
+                                                          domain_proxy_objects):
+    """Executes the steps from Step1 to Step22 for MCP and XPR testcases
+
+    Args:
+       test_type: A string which indicates the testcase type
+       iteration_content: A dictionary with multiple key-value pairs that contain iteration data
+       sas_test_harness_objects: SAS Test Harnesses objects
+       domain_proxy_objects: Domain proxy objects
+    """
+    fad_test_harnesses_objects = []
+    cbsds_with_domain_proxies = []
```

Comment:
Comment what these variables are used for and/or move them down to the last place they could reasonably be created.

Reply:
done as suggested.